src/GafferArnold/IECoreArnoldPreview/Renderer.cpp

```diff
@@ -813,6 +816,16 @@ class ArnoldAttributes : public IECoreScenePreview::Renderer::AttributesInterfac
 			{
 				AiNodeSetByte( node, "visibility", m_visibility );
 				AiNodeSetByte( node, "sidedness", m_sidedness );
+				
+				if( m_transformType )
+				{
+					// \todo : Arnold quite explicitly discourages constructing AtStrings repeatedly,
+					// but given the need to pass m_transformType around as a string for consistency
+					// reasons, it seems like there's not much else we can do here.
+					// Long run, it seems like a lot of things would get cleaner if we had first
```

Comment:
This class is destined for Cortex, and has nothing to do with enum plugs at all - it's a whole level lower than that. Why do you say "there's not much else we can do here" - did the AiEnum query idea not work out?

Also, note that the commit message is out of sync with the code here - it's still describing the previous approach. While on the subject of commit messages, it'd be great if you could keep the subject line short enough to avoid line wrapping, and put any extra information in the body - it makes the logs and the review pages much easier to read.

Reply:
I tried to figure out how the AiEnum approach would work, but it looks like Renderer::ArnoldAtttributes gets repeatedly constructed for each location anyway - as long as the public interface Renderer::attributes gets called for every location, and that takes strings for this property, it seems like we can't really do any better here  ( doing a linear search against a vector of strings probably would be a little bit faster, but would be pretty ugly, and still not particularly optimal ).

The comment on enum plugs comes from talking to Don about it, and him pointing out that aside from performance, a string really isn't a very clean representation of something that can only have 3 values.  Currently, if someone enters this string, and misspells it, we don't flag that until it hits the Arnold ( if I recall correctly, Arnold's error message for this case isn't terrible, but it still seems less than ideal. )  So, anyway, I've kind of given up on anything for the short term, but if this happens often, something cleaner would be nice.

I can scrap the comment if you prefer.  As for mentioning plugs - yes, if we were going to do it, the plug would have to be backed by some sort of Cortex type, and that's what the renderer backend would see, but I don't see much problem mentioning plugs in a comment - is there any point in pretending that anyone cares about Cortex renderer backends in any context other than Gaffer?